test/base_test.py

```diff
@@ -52,10 +52,9 @@ def test_attr_on_root(self, server):
         retrieved_attributes = decode_response(response)
         assert retrieved_attributes == nx_attributes
 
-    @pytest.mark.parametrize("format_arg", ("json", "npy"))
-    @pytest.mark.parametrize("flatten", (False, True))
-    def test_data_on_array(self, server, format_arg, flatten):
-        """Test /data/ endpoint on array dataset in a group"""
+    @pytest.mark.parametrize("format_arg", ("json", "bin", "npy", "csv", "tiff"))
+    def test_data_on_array_with_format(self, server, format_arg):
+        """Test /data/ endpoint on array dataset"""
```

Comment:
This one now tests all the supported formats without setting the other parameters. This means we no longer need to test `flatten: False` and `dtype: 'origin'` in subsequent tests.